chimera_app/steam_config.py

```diff
@@ -0,0 +1,199 @@
+            config_file = user_dir + '/config/localconfig.vdf'
+
+            ensure_directory_for_file(config_file)
+
+            config_data = self.load_local(config_file)
+
+            # Add global entries if file exists
+            self.apply_to_local_config(sc.MAIN_TWEAKS_FILE, config_data)
+            # Add local entries if file exists
+            self.apply_to_local_config(sc.LOCAL_TWEAKS_FILE, config_data)
+
+            self.write(config_data, config_file)
+
+    @staticmethod
+    def write(config_data, config_file):
+        vdf.dump(config_data, open(config_file, 'w'), pretty=True)
+
+    @staticmethod
+    def load_main(path):
+        if file_exists(path):
+            data = vdf.load(open(path))
+        else:
+            data = vdf.VDFDict()
+            data['InstallConfigStore'] = {'Software': {'Valve': {'Steam': {}}}}
+
+        steam = data['InstallConfigStore']['Software']['Valve']['Steam']
+
+        if 'CompatToolMapping' not in steam:
+            steam['CompatToolMapping'] = {}
+        else:
+            stale_entries = []
+
+            for game in steam['CompatToolMapping']:
+                if ('name' in steam['CompatToolMapping'][game] and
+                        'config' in steam['CompatToolMapping'][game]):
+                    if (steam['CompatToolMapping'][game]['name'] == '' and
+                            steam['CompatToolMapping'][game]['config'] == ''):
+                        stale_entries.append(game)
+
+            for entry in stale_entries:
+                del steam['CompatToolMapping'][entry]
+
+        return data
+
+    @staticmethod
+    def load_local(path):
+        if file_exists(path):
+            data = vdf.load(open(path))
+        else:
+            data = vdf.VDFDict()
+            data['UserLocalConfigStore'] = {'Software': {'Valve': {'Steam': {}}}}
+
+        steam_input = data['UserLocalConfigStore']
+        if 'Apps' not in steam_input:
+            steam_input['Apps'] = {}
+
+        launch_options = data['UserLocalConfigStore']['Software']['Valve']['Steam']
+        if 'Apps' not in launch_options:
+            launch_options['Apps'] = {}
+
+        return data
+
+    @staticmethod
+    def download_tweaks_file(file_path, num_atempts=5, static_file=None):
+        """Downloads the latest tweak file from our server. It will try
+        num_atempts or fail. If the file can't be reached after num_atempts
+        then it will be replaced by a static file designated by static_file
+        unless the file already exists.
+
+        returns True if the procedure was successful by downloading the new
+        file or by replacing with static_file
+        """
+        url = ('https://raw.githubusercontent.com/ChimeraOS/'
+               'chimera/master/steam-tweaks.yaml')
+        ret_val = False
+        attempts = 0
+        while (attempts < num_atempts):
+            try:
+                r = requests.get(url, timeout=1)
+            except requests.Timeout:
```

Comment:
If a different kind of error happens here it will kill the script. We don't want that. Any kind of error is fine to skip. Updating the file is a best effort.

Reply:
I would like to continue if there is a Timeout but ignore everything else. I'll add that missing part.